image_recognition_color_extractor/scripts/get_colors_stream

```diff
@@ -7,31 +7,25 @@ from sensor_msgs.msg import Image
 
 from image_recognition_msgs.srv import Recognize
 
-def extract_color_client(img_msg):
-
-    rospy.wait_for_service('extract_color')
-    try:
-        extract_color = rospy.ServiceProxy('extract_color', Recognize)
-        return extract_color(img_msg)
-    except rospy.ServiceException as e:
-        print("Service call failed: %s"%e)
-
-# Create a callback function for the subscriber.
-def callback(msg):
-
-    # Simply print out values in our custom message.
-    colors = extract_color_client(msg)
-    rospy.loginfo(colors)
+class ExtractColorClient(object):
+    def __init__(self, image_topic, color_service):
+        self.image_sub = rospy.Subscriber(image_topic, Image, self.color_callback)
+        self.color_proxy = rospy.ServiceProxy(color_service, Recognize)
+        self.color_proxy.wait_for_service(timeout=20)
```

Comment:
```suggestion
class ExtractColorClient:
    def __init__(self, image_topic, color_service):
        self.color_proxy = rospy.ServiceProxy(color_service, Recognize)
        self.color_proxy.wait_for_service(timeout=20)

        self.image_sub = rospy.Subscriber(image_topic, Image, self.color_callback)
```
As the subscriber will start immediately, but the service client will not be ready yet. So that could lead to an error.